### ISSUE TITLE
memory problems in -prod

### ISSUE THREAD

**Issue by willkg** [mozilla/standup#290]
Running Standups on Heroku has the problem where the process consumes memory over time until it passes the threshold for how much we're allowed for the dyno. Then it restarts and we loop again.

This issue covers looking into memory usage, identifying what thing(s) are consuming memory, and fixing them.

**Comment by willkg** [comment]
**Is it cache?** No.

Today I looked at caching. We use locmem so all caching is done in memory. I theorized that maybe our cache wasn't limited, so it'd grow ad infinitum until we hit memory issues.

Towards that theory, I added some guards to caching in https://github.com/mozilla/standup/commit/bc8b1aedf19c5b7cce0f47db833e6745cacc2e93

If the theory held, then we wouldn't see the normal climb of memory usage.

However, after deploying this, a couple of hours later, we saw the problematic memory usage in the graph, so I don't think this theory holds.

**Comment by pmac** [comment]
To really put a nail in it we could try setting the CACHE_URL to `dummy://`, which effectively disables the cache. If that doesn't help then that's for-sure not it.

**Comment by willkg** [comment]
I'm spending some quality time with this today.

I think the symptom statement is this:

> Over time, the Standup web process increases in RSS memory and response time as it handles incoming HTTP requests. This continues to occur until the process is restarted.

I can see both of these using the Heroku dashboard. New Relic also shows 
We're definitely seeing that in production. We're also being spidered by several bots, so the degradation happens pretty fast.

I wrote a spidering script and if I run it against our stage environment, I'm pretty sure I see the same thing based on graphs on Heroku dashboard.

If I run that spidering script locally, I also see the same thing: RSS memory for the process and the time it takes to handle a request go up. I made sure to set `DEBUG=False` when running locally.

I set the `CACHE_URL` to `dummy:///` as @pmac suggested. I'm seeing the same thing. Given that, I think it's safe to rule out the Django caching system.

I'll add more as I discover it.
